Added create_static method to Scene class.

tarzzz opened the pull request:
It allows you to create static webpages for the visualization.
Added cleanup_static method to remove the static data generated!

**To test this branch**
modify last line in <code>pydy_examples/illustrative_example_for_viz/illustrative_example.py</code>
from:
<code> scene.display()</code>
to:
<code> scene.create_static()</code>
and run it:
<code> python illustrative_example.py</code>

then cd to static directory generated, and from terminal
<code>$ python -m SimpleHTTPServer</code>
You should be able to see visualization at localhost:8000

tarzzz committed:
Added create_static method to Scene class. It allows you to create
static webpages for the visualization.
Added cleanup_static method to remove the static data generated!

tarzzz commented:
@moorepants Please review and merge it. 
P.S.: it was already partially implemented in scene class, but was redundant earlier. Now we can have a better use of it :)

moorepants commented:
This works. No need to open a server. Simply open index.html in your browser.

Gonna add a couple of things before merging.

moorepants committed:
Prevented unwanted file deletion and improve method names.

Safety code is in place in create_static_html and remove_static_html to prevent
unwanted deletions. I also changed the names of the methods to indicate more
precisely what the methods do.

moorepants commented:
These really need tests before merging. Not sure how long that would take. @tarzzz, are you fine with the changes I made.

tarzzz commented on pydy_viz/scene.py:
```diff
@@ -316,24 +317,84 @@ def generate_visualization_json(self, dynamic_variables,
-        distutils.dir_util.remove_tree(os.path.join(os.getcwd(), '.pydy_viz'))
+        dst = os.path.join(os.getcwd(), 'static')
+
+        if os.path.exists(dst) and overwrite is False:
```
I am fine with the changes, but I feel we should be writing tests too for the same(because of conditional statements). I can write them and if we are short of time(sending tutorial instructions to students), then we can merge it and tests can be sent in a separate PR.

moorepants replied:
This isn't critical to have for the tutorial, because we already have at least one browser/os combination that works for everyone. But if this was in a new pydy-viz release before the april 2, then we would have it available for backup in case the socket server isn't working for some.

moorepants commented:
@tarzzz You need to get this merged into the pydy repo.

tarzzz commented:
I am fine with the changes, but I feel we should be writing tests too for
the same(lots of conditional statements). I can write them and if we are
short of time(sending tutorial instructions to students), then we can merge
it and tests can be sent in a separate PR.

On Mon, Mar 31, 2014 at 7:37 AM, Jason K. Moore notifications@github.comwrote:

> These really need tests before merging. Not sure how long that would take.
> @tarzzz https://github.com/tarzzz, are you fine with the changes I made.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/pydy/pydy-viz/pull/117#issuecomment-39048537
> .

tarzzz commented:
Sorry, that was a draft I made some time ago(before PyCon). I am onto it
now.

On Wed, Apr 23, 2014 at 1:22 AM, TARUN GABA tarun.gaba7@gmail.com wrote:

> I am fine with the changes, but I feel we should be writing tests too for
> the same(lots of conditional statements). I can write them and if we are
> short of time(sending tutorial instructions to students), then we can merge
> it and tests can be sent in a separate PR.
> 
> On Mon, Mar 31, 2014 at 7:37 AM, Jason K. Moore notifications@github.comwrote:
> 
> > These really need tests before merging. Not sure how long that would
> > take. @tarzzz https://github.com/tarzzz, are you fine with the changes
> > I made.
> > 
> > —
> > Reply to this email directly or view it on GitHubhttps://github.com/pydy/pydy-viz/pull/117#issuecomment-39048537
> > .

moorepants commented:
@tarzzz 

This would be great to have working in the scipy tutorial. Can you get this working in the new pydy repo?

We plan to send out final instructions by June 23rd so it needs to be done sometime before that.
